lib/rubygem_fs.rb

```diff
@@ -0,0 +1,78 @@
+module RubygemFs
+
+  def self.instance
+    @fs ||=
+      if Rails.env.development?
+        RubygemFs::Local.new
+      else
+        RubygemFs::S3.new
+      end
+  end
+
+  def self.mock!
+    @fs = RubygemFs::Local.new.tap do |fs|
+      def fs.base_dir
+        @dir ||= Dir.mktmpdir
+      end
+    end
+  end
+
+  def self.s3!(host)
+    uri = URI(host)
+    @fs = RubygemFs::S3.new
+    s3 = @fs.s3(access_key_id: 'k', secret_access_key: 's',  proxy: {host: uri.host, port: uri.port})
+    s3.buckets.build(Gemcutter.config['s3_bucket']).save
+  end
+
+  class Local
+    def store(key, body)
+      FileUtils.mkdir_p File.dirname("#{base_dir}/#{key}")
+      File.open("#{base_dir}/#{key}", 'wb') do |f|
```

Comment:
I tried using `File.write`, but I was getting a error: `Encoding::UndefinedConversionError: "\x9C" from ASCII-8BIT to UTF-8`
I think it is because the file was not opened as `wb` which is binary file.